Sources/FoundationEssentials/Calendar/Calendar_Gregorian.swift

```
@@ -10,26 +10,108 @@
+    
+    let julianCutoverDay: Int// Julian day (noon-based) of cutover
+    let gregorianStartYear: Int
+    let gregorianStartDate: Date
+
+    // Only respects Gregorian identifier
     init(identifier: Calendar.Identifier, timeZone: TimeZone?, locale: Locale?, firstWeekday: Int?, minimumDaysInFirstWeek: Int?, gregorianStartDate: Date?) {
-        fatalError()
+
+        self.timeZone = timeZone ?? .gmt
+        if let gregorianStartDate {
+            self.gregorianStartDate = gregorianStartDate
+            self.julianCutoverDay = gregorianStartDate.julianDay
+            let (y, _, _) = Self.yearMonthDayFromJulianDay(julianCutoverDay, useJulianRef: false)
+            self.gregorianStartYear = y
+        } else {
+            self.gregorianStartYear = 1582
+            self.julianCutoverDay = 2299161
+            self.gregorianStartDate = Date(timeIntervalSince1970: -12219292800) // 1582-10-15T00:00:00Z
+        }
+
+        self.locale = locale
+        self.localeIdentifier = locale?.identifier ?? ""
+        if let firstWeekday, (firstWeekday >= 1 && firstWeekday <= 7) {
+            _firstWeekday = firstWeekday
+        }
+
+        if var minimumDaysInFirstWeek {
+            if minimumDaysInFirstWeek < 1 {
+                minimumDaysInFirstWeek = 1
+            } else if minimumDaysInFirstWeek > 7 {
+                minimumDaysInFirstWeek = 7
+            }
+            _minimumDaysInFirstWeek = minimumDaysInFirstWeek
+        }
     }
     
     var identifier: Calendar.Identifier {
         .gregorian
     }
     
     var locale: Locale?
-    
+
     var localeIdentifier: String
     
     var timeZone: TimeZone
     
-    var firstWeekday: Int
-    
-    var minimumDaysInFirstWeek: Int
-    
+    var _firstWeekday: Int?
+    var firstWeekday: Int {
+        set {
+            if newValue >= 1 && newValue <= 7 {
+                _firstWeekday = newValue
+            }
```

Is there any value in a `fatalError` here instead of silently dropping the value?